src/Data/Parameterized/List.hs

```diff
@@ -51,8 +51,11 @@ data List :: (k -> *) -> [k] -> * where
 infixr 5 :<
 
 instance ShowF f => Show (List f sh) where
-  show Nil = "Nil"
-  show (elt :< rest) = showF elt ++ " :< " ++ show rest
+  showsPrec _ Nil = showString "Nil"
+  showsPrec p (elt :< rest) = showParen (p > precCons) $
+    showsPrecF (precCons+1) elt . showString " :< " . showsPrec (precCons+1) rest
```

Comment:
Could you modify this to take advantage of right-associativity of ":<" and reduce the number of parens?

To see how Haskell does this for String, see the definition of "showList__" in https://hackage.haskell.org/package/base-4.10.1.0/docs/src/GHC.Show.html.

Reply:
Done. Note that derived `Show` and `Read` instances don't eliminate redundant parens implied by associativity, and derived `Read` instances fail to parse if they are omitted: http://hackage.haskell.org/package/base-4.10.1.0/docs/Prelude.html#t:Show. But of course there are no derived `Read` instances here, so we can do whatever we want.